working/augmentation-libraries/feature-specification.md

```diff
@@ -321,82 +322,150 @@ doesn't, it must repeat the type from the augmented definition__.
+    of the augmented declaration.
+
+*   **Augmenting operators**: Operators are methods with special
+    invocation syntax. When augmenting an operator declaration, `augmented`
+    refers to the augmented operator *method*, and it must occur only
+    as the function of a function invocation. For example, when augmenting
     `operator +` you could use `augmented(1)` to call the augmented
-    operator, and when augmenting `operator []=` you would use the
-    `augmented(key, value)` syntax.
-
-    *   Note that `augmented` in such an augmenting operator method body is
-        not an expression by itself, and cannot be used to tear off the
-        augmented operator method. Similar to `super`, it is a syntactic
-        form which can only be used in limited ways.
+    operator, when augmenting `operator []=` you would use the
+    `augmented(key, value)` syntax, and when augmenting `operator -()`,
+    you would use `augmented()`.
+    Parsing, evaluation and type inference is handled like a
+    function augmentation.
 
 *   **Augmenting enum values**: When augmenting an enum value, `augmented`
-    has no meaning and is not allowed.
+    has no meaning and is not allowed to occur in constructor argument
+    expressions.
+
+*   **Augmenting constructors**:
+    *   **Non-redirecting generative constructor**:
+        The term `augmented` has no meaning, and must not occur in
+        optional parameter default value expressions, initializer list
+        entry expressions (whether initializing or asserts),
+        super-constructor invocation argument expressions, or in the
+        body. _Non-redirecting generative constructors automatically
+        invoke their augmented constructor after executing their
+        initializer list, and will execute their body after that of
+        their augmented constructor._
+    *   **Non-redirecting Factory constructor**: An `augmented` in the
+        constructor body refers to the augmented non-redirecting factory
+        constructor. It must occur only as the function of a (non-generic)
+        function invocation.
+        The term <code>'augmented' \<argumentList\></code> clause parses
+        as a `<primary>` expression.
+        When evaluated, it invokes the augmented constructor on the same
+        class and evaluates to the returned value.
+        Type inference is performed as for invoking a constructor with
+        the same function type.
+        The term `augmented` must not occur in optional parameter
+        default value expressions.
+    *   **Redirecting generative constructor**:
+        The term `augmented` has no meaning, and must not occur in
+        optional parameter default value expressions or in
+        target constructor arguments expressions.
+    *   **Redirecting factory constructor**: Contains no expressions.
+
+    The term <code>'augmented' \<argumentPart\></code> words parses
+    as a `<primary>` expression.
```

Comment:
Is this leftover text? If so, delete. If not... what's it about? At the very least, I think you can delete "words".

Reply:
I think it's deliberate, and it's really saying that `augmented(args)` is a `<primary>` inside an augmenting function, not a `<primary><selector>` parsed as a `<postfixExpression>`.

It's tricky. It's really a way of saying that we have a parameterized grammar that is unique to each kind of allowed `augmented` syntax. We do not allow parsing `augmented` as a `<primary>` inside an augmenting function or setter, only inside an augmenting variable or getter. At most one of the `augmented` structures is *enabled* at a time.
Our grammar doesn't really allow that (and I don't want five copies of all the rules from `<expression>` down to `<primary>`, so I'm trying to say it with words.
It's like a disambiguation rule: Inside this kind of augmenting declaration, use this production (only).

It's clearly not clear :)